experimental/devices/mcp9808/mcp9808.go

```diff
@@ -0,0 +1,426 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package mcp9808
+
+import (
+	"encoding/binary"
+	"errors"
+	"sync"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Opts holds the configuration options.
+//
+// Slave Address
+//
+// Depending which pins the A0, A1 and A2 pins are connected to will change the
+// slave address. Default configuration is address 0x18 (Ax pins to GND). For a
+// full address table see datasheet.
+type Opts struct {
+	Addr int
```

Comment:
uint16 ?

Reply:
Oh, the only reason I do that that way is for the flag be it int or uint since there is no uint16 flag. If its int I can bounds check in the New() can convert to uint16 there. Just makes it cleaner user code imo.
I can change it if you think it better.

Reply:
```go
address := flag.Int("addr", 0x18, "I²C address")
```

```go
// Create a new temperature sensor a sense with default options.
sensor, err := mcp9808.New(bus, &mcp9808.Opts{Addr: *addr})
if err != nil {
	return fmt.Errorf("failed to open new sensor: %v", err)
}
```
rather than 
```go
sensor, err := mcp9808.New(bus, &mcp9808.Opts{Addr: uint16(*addr)})
if err != nil {
	return fmt.Errorf("failed to open new sensor: %v", err)
}
```

Reply:
I started thinking about adding a i2c.Addr type that would implement Set(). Then it'd be easy to use as a flag.

Yes, I prefer uint16 for i2c address.